localstack/utils/testutil.py

```diff
@@ -125,7 +125,7 @@ def create_lambda_archive(
 
 
 def delete_lambda_function(name, region_name: str = None):  # TODO: remove all occurrences
-    client = aws_stack.connect_to_service("lambda", region_name=region_name)
+    client = connect_to(region_name=region_name).awslambda
```

Comment:
In the test utils we should always use connect_externally_to.

Reply:
Changed in 250caa25fb8336c9293dcd622b4548d151f7ad34.

This is marked for removal anyway, I'll do it in a subsequent PR.